src/development/tools/devtools/performance.md

```diff
@@ -158,6 +158,16 @@ enable the Track Paints option:
 
 ## More debugging options
 
+To diagnose performance problems related to rendering layers,
+toggle off a rendering layer.
+These options are enabled by default.
+
+To see the effects on your app's performance,
+reproduce the activity in your app,
+select the new frames in the frames chart,
+to inspect the timeline events
+with the layers disabled.
```

Comment:
maybe add a sentence here that says. "If Paint time or Raster time has significantly decreased, excessive use of the effects you disabled may be contributing to the jank you saw in your app."

@dnfield can you check me here - if a user disables clip layers and sees a performance improvement, where is that improvement likely to show up? In the Paint event on the UI thread? In the raster events?

Reply:
Raster.

Reply:
thanks @dnfield . and is this true for all the layer options here? (clip, opacity, physical shape)

Reply:
In general, yes - although _some_ build time improvements on the UI thread may be noted, these are things that have more impact on raster time.

Reply:
(i.e. clipping (especially clipping with anti-aliasing if it is requested), saving layers, drawing shadows)

Reply:
added a sentence for this in [fe1a818](https://github.com/flutter/website/pull/6975/commits/fe1a818936387c4d33d09b18f3f6fc95edfa9563)